src/libraries/System.Private.CoreLib/src/System/Reflection/InvokeUtils.cs

```diff
@@ -123,5 +123,181 @@ private static bool TryConvertPointer(object srcObject, [NotNullWhen(true)] out
+                            Debug.Fail("Array.Copy from U4 to another type hit unsupported widening conversion"); break;
+                    }
+                    break;
+
+                case CorElementType.ELEMENT_TYPE_I4:
+                    switch (destElType)
+                    {
+                        case CorElementType.ELEMENT_TYPE_I8:
+                            Unsafe.As<byte, long>(ref destElement) = Unsafe.As<byte, int>(ref srcElement); break;
+                        case CorElementType.ELEMENT_TYPE_R4:
+                            Unsafe.As<byte, float>(ref destElement) = Unsafe.As<byte, int>(ref srcElement); break;
+                        case CorElementType.ELEMENT_TYPE_R8:
+                            Unsafe.As<byte, double>(ref destElement) = Unsafe.As<byte, int>(ref srcElement); break;
+                        default:
+                            Debug.Fail("Array.Copy from I4 to another type hit unsupported widening conversion"); break;
+                    }
+                    break;
+
+                case CorElementType.ELEMENT_TYPE_U8:
+                    switch (destElType)
+                    {
+                        case CorElementType.ELEMENT_TYPE_R4:
+                            Unsafe.As<byte, float>(ref destElement) = Unsafe.As<byte, ulong>(ref srcElement); break;
+                        case CorElementType.ELEMENT_TYPE_R8:
+                            Unsafe.As<byte, double>(ref destElement) = Unsafe.As<byte, ulong>(ref srcElement); break;
+                        default:
+                            Debug.Fail("Array.Copy from U8 to another type hit unsupported widening conversion"); break;
+                    }
+                    break;
+
+                case CorElementType.ELEMENT_TYPE_I8:
+                    switch (destElType)
+                    {
+                        case CorElementType.ELEMENT_TYPE_R4:
+                            Unsafe.As<byte, float>(ref destElement) = Unsafe.As<byte, long>(ref srcElement); break;
+                        case CorElementType.ELEMENT_TYPE_R8:
+                            Unsafe.As<byte, double>(ref destElement) = Unsafe.As<byte, long>(ref srcElement); break;
+                        default:
+                            Debug.Fail("Array.Copy from I8 to another type hit unsupported widening conversion"); break;
+                    }
+                    break;
+
+                case CorElementType.ELEMENT_TYPE_R4:
+                    Debug.Assert(destElType == CorElementType.ELEMENT_TYPE_R8);
+                    Unsafe.As<byte, double>(ref destElement) = Unsafe.As<byte, float>(ref srcElement); break;
+
+                default:
+                    Debug.Fail("Fell through outer switch in PrimitiveWiden!  Unknown primitive type for source array!"); break;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool CanPrimitiveWiden(CorElementType srcET, CorElementType dstET)
```

Comment:
Please comment what this code is trying to do and what the look-up table is encoding.

Reply:
For example `ELEMENT_TYPE_I8` is marked as `0x3400`, but it's real value is `0x0a`. That is confusing.

Reply:
The NativeAot and native counterparts of this don't include comment for this. `ELEMENT_TYPE_I8` here is the index instead of value. It's effectively an `ELEMENT_TYPE_I8 => 0x3400` arm.

Reply:
> The NativeAot and native counterparts of this don't include comment for this. `E`

Not sure that is an acceptable defense in this case. I genuinely don't know what this is doing and for new code that I am reviewing, I would prefer to have it documented in some manner.

Reply:
Actually. Stand by on this. I am seeing your concern now.

Reply:
I still feel the comment too verbose for corelib contributor. A simpler comment would be nice.

Reply:
What would you propose that captures the use and is shorter?

Reply:
Ah, I see your update.